pkg/controller/secret_control.go

```diff
@@ -0,0 +1,140 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	certutil "github.com/pingcap/tidb-operator/pkg/util/crypto"
+	corev1 "k8s.io/api/core/v1"
+	types "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+)
+
+// SecretControlInterface manages certificates used by TiDB clusters
+type SecretControlInterface interface {
+	Create(certOpts *TiDBClusterCertOptions, cert []byte, key []byte) error
+	Load(ns string, secretName string) ([]byte, []byte, error)
+	Check(ns string, secretName string) bool
+}
+
+type realSecretControl struct {
+	kubeCli      kubernetes.Interface
+	secretLister corelisters.SecretLister
+}
+
+// NewRealSecretControl creates a new SecretControlInterface
+func NewRealSecretControl(
+	kubeCli kubernetes.Interface,
+	secretLister corelisters.SecretLister,
+) SecretControlInterface {
+	return &realSecretControl{
+		kubeCli:      kubeCli,
+		secretLister: secretLister,
+	}
+}
+
+func (rsc *realSecretControl) Create(certOpts *TiDBClusterCertOptions, cert []byte, key []byte) error {
+	secretName := fmt.Sprintf("%s-%s", certOpts.Instance, certOpts.Suffix)
+
+	secret := &corev1.Secret{
+		ObjectMeta: types.ObjectMeta{
+			Name:   secretName,
+			Labels: make(map[string]string),
+		},
+		Data: map[string][]byte{
+			"cert": cert,
+			"key":  key,
+		},
+	}
+
+	labelTemp := label.New()
+	secret.Labels[label.NamespaceLabelKey] = certOpts.Namespace
+	secret.Labels[label.ManagedByLabelKey] = labelTemp[label.ManagedByLabelKey]
+	secret.Labels[label.InstanceLabelKey] = certOpts.Instance
+	secret.Labels[label.ComponentLabelKey] = certOpts.Component
+
+	_, err := rsc.kubeCli.CoreV1().Secrets(certOpts.Namespace).Create(secret)
+	glog.Infof("save cert to secret %s/%s, error: %v", certOpts.Namespace, secretName, err)
+	return err
+}
+
+// Load loads cert and key from Secret matching the name
+func (rsc *realSecretControl) Load(ns string, secretName string) ([]byte, []byte, error) {
+	secret, err := rsc.secretLister.Secrets(ns).Get(secretName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return secret.Data["cert"], secret.Data["key"], nil
+}
+
+// Check returns true if the secret already exist
+func (rsc *realSecretControl) Check(ns string, secretName string) bool {
+	certBytes, keyBytes, err := rsc.Load(ns, secretName)
+	if err != nil {
+		return false
+	}
+
+	// validate if the certificate is valid
+	block, _ := pem.Decode(certBytes)
```

Comment:
don't omit `err`

Reply:
It's not an error `func Decode(data []byte) (p *Block, rest []byte)`